packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -268,6 +268,7 @@ export const DatePicker = ({
   return (
     <FocusOn
       scrollLock={false}
+      returnFocus={false}
```

Comment:
The update caused `returnFocus` to run despite us manually handling `onDeactivation`, which lead to inaccurate focusing.